spring-cloud-stream/src/main/java/org/springframework/cloud/stream/converter/KryoMessageConverter.java

```diff
@@ -0,0 +1,263 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.converter;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import com.esotericsoftware.kryo.pool.KryoFactory;
+import com.esotericsoftware.kryo.pool.KryoPool;
+import org.objenesis.strategy.StdInstantiatorStrategy;
+
+import org.springframework.integration.codec.kryo.CompositeKryoRegistrar;
+import org.springframework.integration.codec.kryo.KryoRegistrar;
+import org.springframework.lang.Nullable;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.messaging.converter.ContentTypeResolver;
+import org.springframework.messaging.converter.DefaultContentTypeResolver;
+import org.springframework.messaging.converter.MessageConversionException;
+import org.springframework.messaging.converter.SmartMessageConverter;
+import org.springframework.messaging.support.MessageBuilder;
+import org.springframework.messaging.support.MessageHeaderAccessor;
+import org.springframework.util.Assert;
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.MimeType;
+
+/**
+ * @author Vinicius Carvalho
+ */
+public class KryoMessageConverter implements SmartMessageConverter {
+
+	protected final KryoPool pool;
+
+	private final CompositeKryoRegistrar kryoRegistrar;
+
+	private final boolean useReferences;
+
+	private ConcurrentMap<String, MimeType> mimeTypesCache = new ConcurrentHashMap<>();
+
+	private final List<MimeType> supportedMimeTypes;
+
+	public static final String KRYO_MIME_TYPE = "application/x-java-object";
+
+	private ContentTypeResolver contentTypeResolver = new DefaultContentTypeResolver();
+
+
+
+	public KryoMessageConverter(List<KryoRegistrar> kryoRegistrars, boolean useReferences) {
+
+		this.useReferences = useReferences;
+		this.kryoRegistrar = CollectionUtils.isEmpty(kryoRegistrars) ? null :
+				new CompositeKryoRegistrar(kryoRegistrars);
+
+		KryoFactory factory = () -> {
+			Kryo kryo = new Kryo();
+			configureKryoInstance(kryo);
+			return kryo;
+		};
+		this.pool = new KryoPool.Builder(factory).softReferences().build();
+		this.supportedMimeTypes = Collections.singletonList(MimeType.valueOf(KRYO_MIME_TYPE));
+	}
+
+	@Nullable
+	@Override
+	public Object fromMessage(Message<?> message, Class<?> targetClass, @Nullable Object conversionHint) {
+
```

Comment:
please avoid inserting empty lines inside the method.

Reply:
Also, there are unnecessary empty lines (more than 1) before the ctor.

Reply:
Well, it depends, @ilayaperumalg .

If your method declaration wrap its arguments to a new line, the good practice indeed to insert blank line after that for better readability. But not in this case, agreed.